client/src/components/ElementActions/ArchiveAction.js

```diff
@@ -0,0 +1,46 @@
+/* global confirm */
+import React from 'react';
+import { compose } from 'redux';
+import AbstractAction from 'components/ElementActions/AbstractAction';
+import archiveBlockMutation from 'state/editor/archiveBlockMutation';
+import i18n from 'i18n';
+
+/**
+ * Adds the elemental menu action to archive a block of any state
+ */
+const ArchiveAction = (WrappedComponent) => (props) => (
```

Comment:
The reason I chose this is because it's heavily used as an example in the [React HOCs documentation](https://reactjs.org/docs/higher-order-components.html)

Reply:
Resolved -> renamed to MenuComponent as suggested further down in the feedback